src/main/java/com/github/alinaberlin/security/components/JWTEncoderDecoder.java

```diff
@@ -0,0 +1,87 @@
+package com.github.alinaberlin.security.components;
+
+import com.github.alinaberlin.security.models.User;
+import io.jsonwebtoken.*;
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class JWTEncoderDecoder {
+    private final String secret_key = "mysecretkeyisthesecretkeymysecretkeyisthesecretkey";
```

Comment:
When we define constants they should be "static final" and for the naming convention it should be snake case with all uppercase. So it would look something like this: `private final String secret_key = "SECRET";`. Also since this is a secret this shouldn't be hardcoded. Take a look at https://www.youtube.com/watch?v=PmGLn3ua_lU&pp=ygUTc3ByaW5nIGJvb3Qgc2VjcmV0cw%3D%3D